File: java/src/main/java/ch3_recursion/ArrLength.java
package ch3_recursion;

public class ArrLength {
    public static void main(String[] args) {
        int[] a = {1, 4, 10, 18, 2, -4};
        System.out.print(arrLength(a));
    }

    public static int arrLength(int[] arr) {
        return arrLength(arr, 0);
    }

    private static int arrLength(int[] arr, int index) {
        try {
            // Try to access the element at index and if successful continue on
            int temp = arr[index];
            return arrLength(arr, index + 1);
        } catch (ArrayIndexOutOfBoundsException e) {
            return index;
        }
    }
}

File: python/03_recursion/arr_length.py
def arr_length(arr, index=0):
    # return length of list recursively
    try:
        temp = arr[index]
        return arr_length(arr, index + 1)
    except IndexError:
        return index


if __name__ == "__main__":
    print(arr_length([5, 4, 3, 2, 10, 80, 11]))
